common/src/main/java/com/wynntils/gui/screens/settings/WynntilsBookSettingsScreen.java

```diff
@@ -0,0 +1,224 @@
+            AbstractButton featureButton = features.get(i);
+            featureButton.render(poseStack, mouseX, mouseY, partialTick);
+        }
+
+        featureListScrollButton.renderButton(poseStack, mouseX, mouseY, partialTick);
+    }
+
+    private static void renderScrollArea(PoseStack poseStack) {
+        RenderSystem.enableBlend();
+        RenderUtils.drawTexturedRect(
+                poseStack,
+                Texture.SETTING_SCROLL_AREA,
+                (Texture.SETTING_BACKGROUND.width() / 2f - Texture.SETTING_SCROLL_AREA.width()) / 2f,
+                10);
+        RenderSystem.disableBlend();
+    }
+
+    private static void renderBg(PoseStack poseStack) {
+        RenderUtils.drawTexturedRect(poseStack, Texture.SETTING_BACKGROUND, 0, 0);
+    }
+
+    // endregion
+
+    // region mouse events
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+        mouseX -= getTranslationX();
+        mouseY -= getTranslationY();
+
+        for (GuiEventListener child : children()) {
+            if (child.isMouseOver(mouseX, mouseY)) {
+                child.mouseClicked(mouseX, mouseY, button);
+            }
+        }
+
+        for (int i = scrollOffset; i < Math.min(features.size(), scrollOffset + FEATURES_PER_PAGE); i++) {
+            AbstractButton featureButton = features.get(i);
+            if (featureButton.isMouseOver(mouseX, mouseY)) {
+                featureButton.mouseClicked(mouseX, mouseY, button);
+            }
+        }
+
+        if (featureListScrollButton.isMouseOver(mouseX, mouseY)) {
+            featureListScrollButton.mouseClicked(mouseX, mouseY, button);
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean mouseScrolled(double mouseX, double mouseY, double delta) {
+        mouseX -= getTranslationX();
+        mouseY -= getTranslationY();
+
+        if (mouseX <= Texture.SETTING_BACKGROUND.width() / 2f) {
+            featureListScrollButton.mouseScrolled(mouseX, mouseY, delta);
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean mouseReleased(double mouseX, double mouseY, int button) {
+        if (featureListScrollButton.isMouseOver(mouseX, mouseY)) {
+            featureListScrollButton.mouseReleased(mouseX, mouseY, button);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean mouseDragged(double mouseX, double mouseY, int button, double dragX, double dragY) {
+        featureListScrollButton.mouseDragged(mouseX, mouseY, button, dragX, dragY);
+
+        return true;
+    }
+
+    private void scroll(double delta) {
+        scrollOffset =
+                MathUtils.clamp((int) (scrollOffset - delta), 0, Math.max(0, features.size() - FEATURES_PER_PAGE));
+    }
+
+    // endregion
+
+    @Override
+    public void onClose() {
+        ConfigManager.loadConfigFile();
+        ConfigManager.loadConfigOptions(ConfigManager.getConfigHolders(), true);
```

Comment:
This should only be called when the "close" button is called, and not when the config is saved.

Reply:
We do this with overlays too. We save and then reload the current config state. Perhaps this is not needed though. Idk why I did this the first time.

Reply:
I though about this and I like the way it is now. We re-load every config option, to update their states.